server/routes/bookings/bookingDetailsController.ts

```diff
@@ -26,6 +25,7 @@ export default class BookingDetailsController {
 
       const { visitDisplayId } = matchedData<{ visitDisplayId: string }>(req)
 
+      const { visits } = bookings
```

Comment:
Unrelated, but spotted pre-existing bug here (and also present in cancel controllers). `bookings` may not be present (e.g. reloading from an expired session) so need to check for validation errors **before** trying to restructure `visits` from it.